app/code/core/Mage/Core/Model/Cookie.php

```diff
@@ -236,8 +252,29 @@ public function set($name, $value, $period = null, $path = null, $domain = null,
         if (is_null($httponly)) {
             $httponly = $this->getHttponly();
         }
+        if (is_null($sameSite)) {
+            $sameSite = $this->getSameSite();
+        }
 
-        setcookie($name, $value, $expire, $path, $domain, $secure, $httponly);
+        if (PHP_VERSION_ID >= 70300) {
+            setcookie(
+                $name,
+                $value,
+                [
+                    'expires'  => $expire,
+                    'path'     => $path,
+                    'domain'   => $domain,
+                    'secure'   => $secure,
+                    'httponly' => $httponly,
+                    'samesite' => $sameSite
+                ]
+            );
+        } else {
+            if (!empty($sameSite)) {
+                $path.= "; samesite=${$sameSite}";
```

Comment:
Function getSameSite() should not be able to return an empty string. It's a configurable by a dropdown, with a default value defined in config.xml. If for whatever reason it still might be null, it will force the return of "None" in code.